src/docfx/build/moniker/MonikerProvider.cs

```diff
@@ -81,36 +83,49 @@ public SourceInfo<string> GetConfigMonikerRange(FilePath file)
             return (null, monikers);
         }
 
-        private (Error? error, string[] monikers) GetFileLevelMonikersCore(FilePath file)
+        private (List<Error> errors, string[] monikers) GetFileLevelMonikersCore(FilePath file)
         {
+            var errors = new List<Error>();
             var (_, metadata) = _metadataProvider.GetMetadata(file);
 
             var configMonikerRange = GetConfigMonikerRange(file);
-            var configMonikers = _rangeParser.Parse(configMonikerRange);
+            var configedMonikers = _rangeParser.Parse(configMonikerRange);
 
-            if (!string.IsNullOrEmpty(metadata.MonikerRange))
+            if (!string.IsNullOrEmpty(metadata.MonikerRange) || metadata.Monikers.Length > 0)
             {
                 // Moniker range not defined in docfx.yml/docfx.json,
                 // user should not define it in file metadata
-                if (configMonikers.Length == 0)
+                if (configedMonikers.Length == 0)
                 {
-                    return (Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange), configMonikers);
+                    errors.Add(Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange));
+                    return (errors, configedMonikers);
                 }
 
                 var fileMonikers = _rangeParser.Parse(metadata.MonikerRange);
-                var intersection = configMonikers.Intersect(fileMonikers).ToArray();
+
+                // monikerRagen takes precedence over monikers since it is more likely from user configuration
+                if (fileMonikers.Length > 0 && metadata.Monikers.Length > 0)
```

Comment:
Prefer using _existance_ over _length_ to check if `monikers` or `monikerRange` is defined. Can change the data model to be nullable.

Reply:
what is the difference?

Reply:
A `monikerRange` like `>netcore-2.0 && <netcore-2.0` could evaluate to an empty moniker list, but that doesn't mean a user didn't write it.

Reply:
Actually, we do not need to change the Model type to nullable, just check on `metadta.MonikerRange` would be enough.